python/setup.py

```diff
@@ -4,40 +4,7 @@
 from skbuild import setup
 
 setup(
-    name="rmm",
-    version="23.04.00",
-    description="rmm - RAPIDS Memory Manager",
-    url="https://github.com/rapidsai/rmm",
-    author="NVIDIA Corporation",
-    license="Apache 2.0",
-    classifiers=[
-        "Intended Audience :: Developers",
-        "Topic :: Database",
-        "Topic :: Scientific/Engineering",
-        "License :: OSI Approved :: Apache Software License",
-        "Programming Language :: Python",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
-    ],
-    # Include the separately-compiled shared library
-    extras_require={"test": ["pytest"]},
+    license_files=["LICENSE"],
```

Comment:
We can use [the `license-files` key from Setuptools]( https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration ) for this in `pyproject.toml`

Reply:
Yes, I was going to do that but was there a reason that [you tried this and then removed it in an earlier commit](https://github.com/rapidsai/rmm/pull/1151/commits/14743e6efa88175584d13cd667f55739af70f3e8)?

Reply:
OK, using the license feature here works fine AFAICT.

Reply:
Ah that was a different thing. I was hoping we could use the `license` field to include the metadata and the license file, but it turned out to be either/or. So went with metadata. This is a different field that doesn't overlap. So shouldn't be an issue

Reply:
Ah good catch I missed that. I can include the license = Apache 2.0 for metadata. I think the license-files defaults to the correct behavior anyway, but can add it to be safe since there appears to be some potential room for change on the horizon with PEP 639.

Reply:
Ah ok. Sorry was writing that up as a review comment and just saw your comment here

Reply:
Hmm actually now I'm not quite sure what makes the most sense. Is there a downside to just specifying the file with the `license` key? Note that that key is part of the general PEP 621 spec, whereas license-files is a setuptools extension. The latter allows customizing what filenames to search for (by glob), so it's not an exact path to the file. Is there a reason not to simply do `license = { file = "LICENSE" }`? I think that gets us what we want, and PyPI (or any PyPI-like index) will infer the type of the license from the standard text IIRC.

Reply:
I closed your two newer discussions so that we could continue discussion here.

Reply:
Few thoughts:

* Explicit is better than implicit
* Specifying the metadata can ensure it is [SPDX]( https://spdx.org/licenses/ ) compliant (for [example]( https://spdx.org/licenses/Apache-2.0.html ))
* The metadata here can be extracted and used in the Conda recipe, which would be more DRY friendly

Reply:
Sounds good, I will apply both of the changes then.